### ISSUE TITLE
Back navigation does not work after traversing the first page in a 10 page stack with Shell Navigation

### ISSUE THREAD

**Issue by SoureMaster** [dotnet/maui#24787]
### Description

I am currently using Shell Navigation on the app I am creating. There is a signup process in which the user must complete. Back arrow does not fire after traversing to the second page of a 10 page stack. 

First Page - Landing page
Second Page - First page of sign up process (Back arrow works)
Third Page to 10th page of sign up, back arrow does not fire and the user is stuck.
![Image](https://github.com/user-attachments/assets/a6f0d59b-365d-4a82-8dc8-2f1a121bacff)

### Link to public reproduction project repository

[_MauiNavIssue_](https://github.com/SoureMaster/MauiNavIssue)

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

9.0.0-preview.7.24407.4

### Affected platforms

Android

**Comment by tj-devel709** [comment]
Hi, if you could provide a github link to a repro sample, that would help us in identifying the bug. Thanks!

**Comment by SoureMaster** [comment]
I have added link to repository. Perhaps being new to Maui I am doing something wrong.

**Comment by PureWeen** [comment]
@SoureMaster This one might be fixed on RC2

Does this issue look similar?
https://github.com/dotnet/maui/issues/24555

**Comment by ninachen03** [comment]
This issue has been verified using Visual Studio 17.12.0 Preview 2.0(9.0.0-rc.1.24453.9 &9.0.0-preview.7.24407.4& 8.0.82). Can repro this issue with the [MauiNavIssue](https://github.com/SoureMaster/MauiNavIssue)

**Comment by SoureMaster** [comment]
> @SoureMaster This one might be fixed on RC2
> 
> Does this issue look similar? #24555

Yes. Issue has not been solved in over a year. Not good. @PureWeen PureWeen

**Comment by SoureMaster** [comment]
Yes, it does look similiar. @PureWeen

**Comment by QianaJiao** [comment]
Hi @PureWeen @SoureMaster  
This might be a little different with https://github.com/dotnet/maui/issues/24555
**As for 24787**, it does repro on Android and Windows, it is not a regression, and I found a similar issue here: [stackoverflow 67653489](https://stackoverflow.com/questions/67653489/gotoasync-build-in-back-button-does-not-go-back-to-previous-page-when-working-w) 
OUTPUT: 
Open Page 2 then Open 3/4/5 pages and click back button: 
```
[0:] Microsoft.Maui.Controls.Shell: Warning: Failed to Navigate Back
System.ArgumentException: Ambiguous routes matched for: //IMPL_PageOne/IMPL_PageOne/PageOne/PageTwo matches found: //IMPL_PageOne/IMPL_PageOne/PageOne/PageTwo,//IMPL_PageOne/IMPL_PageOne/PageOne/PageTwo (Parameter 'uri')
   at Microsoft.Maui.Controls.ShellUriHandler.GetNavigationRequest(Shell shell, Uri uri, Boolean enableRelativeShellRoutes, Boolean throwNavigationErrorAsException, ShellNavigationParameters shellNavigationParameters) in /_/src/Controls/src/Core/Shell/ShellUriHandler.cs:line 207
   at Microsoft.Maui.Controls.ShellNavigationManager.GoToAsync(ShellNavigationParameters shellNavigationParameters, ShellNavigationRequest navigationRequest) in /_/src/Controls/src/Core/Shell/ShellNavigationManager.cs:line 63
   at Microsoft.Maui.Controls.ShellSection.NavigationImpl.OnPopAsync(Boolean animated) in /_/src/Controls/src/Core/Shell/ShellSection.cs:line 1061
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellToolbarTracker.OnNavigateBack() in /_/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs:line 228

```

So, I followed the workaround and just deleted "Route=" PageOne ""... then it worked:
![Animation](https://github.com/user-attachments/assets/04cad7db-d1da-4bf1-867f-f878d33783b2)

**Comment by SoureMaster** [comment]
@QianaJiao Thanks, that Stack overflow info and your code solved the issue. Thanks again.

**Comment by jfversluis** [comment]
Looks like this was resolved, thanks all!
